x/evm/keeper/fees_test.go

```diff
@@ -444,99 +473,133 @@ func (suite *KeeperTestSuite) TestVerifyFeeAndDeductTxCostsFromUserBalance() {
+				initialBaseFee = setup.initialBaseFee.BigInt()
+				params := suite.app.FeeMarketKeeper.GetParams(suite.ctx)
+				params.BaseFee = setup.initialBaseFee
+				err := suite.app.FeeMarketKeeper.SetParams(suite.ctx, params)
+				suite.Require().NoError(err)
 
-			if tc.malleate != nil {
-				tc.malleate()
-			}
-			var amount, gasPrice, gasFeeCap, gasTipCap *big.Int
-			if tc.cost != nil {
-				amount = tc.cost.BigInt()
-			}
+				vmdb := suite.StateDB()
 
-			if suite.enableFeemarket {
-				if tc.gasFeeCap != nil {
-					gasFeeCap = tc.gasFeeCap
+				if tc.malleate != nil {
+					tc.malleate()
 				}
-				if tc.gasTipCap == nil {
-					gasTipCap = oneInt.BigInt()
+				var amount, gasPrice, gasFeeCap, gasTipCap, initBal *big.Int
+
+				if tc.cost != nil {
+					amount = tc.cost.BigInt()
+				}
+
+				if suite.enableFeemarket {
+					if tc.gasFeeCap != nil {
+						gasFeeCap = tc.gasFeeCap()
+					}
+					if tc.gasTipCap == nil {
+						gasTipCap = oneInt.BigInt()
+					} else {
+						gasTipCap = tc.gasTipCap
+					}
+					initBal = tc.initialBalance().BigInt()
+
 				} else {
-					gasTipCap = tc.gasTipCap
+					if tc.gasPrice != nil {
+						gasPrice = tc.gasPrice.BigInt()
+					}
+					initBal = hundredInt.BigInt()
 				}
-				vmdb.AddBalance(suite.address, initBalance.BigInt())
-				balance := vmdb.GetBalance(suite.address)
-				suite.Require().Equal(balance, initBalance.BigInt())
-			} else {
-				if tc.gasPrice != nil {
-					gasPrice = tc.gasPrice.BigInt()
+
+				if s.denomDecimals == evmtypes.Denom6Dec {
+					initBal = evmtypes.Convert6To18DecimalsBigInt(initBal)
+					if amount != nil {
+						amount = evmtypes.Convert6To18DecimalsBigInt(amount)
+					}
+					if gasPrice != nil {
+						gasPrice = evmtypes.Convert6To18DecimalsBigInt(gasPrice)
+					}
+					if gasFeeCap != nil {
+						gasFeeCap = evmtypes.Convert6To18DecimalsBigInt(gasFeeCap)
+					}
+					if gasTipCap != nil {
+						gasTipCap = evmtypes.Convert6To18DecimalsBigInt(gasTipCap)
+					}
 				}
 
-				vmdb.AddBalance(suite.address, hundredInt.BigInt())
+				vmdb.AddBalance(suite.address, initBal)
 				balance := vmdb.GetBalance(suite.address)
-				suite.Require().Equal(balance, hundredInt.BigInt())
-			}
-			err := vmdb.Commit()
-			suite.Require().NoError(err, "Unexpected error while committing to vmdb: %d", err)
+				suite.Require().Equal(balance, initBal)
 
-			ethTxParams := &evmtypes.EvmTxArgs{
-				ChainID:   zeroInt.BigInt(),
-				Nonce:     1,
-				To:        &suite.address,
-				Amount:    amount,
-				GasLimit:  tc.gasLimit,
-				GasPrice:  gasPrice,
-				GasFeeCap: gasFeeCap,
-				GasTipCap: gasTipCap,
-				Accesses:  tc.accessList,
-			}
-			tx := evmtypes.NewTx(ethTxParams)
-			tx.From = tc.from
+				err = vmdb.Commit()
+				suite.Require().NoError(err, "Unexpected error while committing to vmdb: %d", err)
 
-			txData, _ := evmtypes.UnpackTxData(tx.Data)
+				ethTxParams := &evmtypes.EvmTxArgs{
+					ChainID:   zeroInt.BigInt(),
```

Comment:
This is different than the one used during the setup:
https://github.com/evmos/evmos/blob/e391d53da336cc678ea108f06f47954aaf1ab866/x/evm/keeper/setup_test.go#L92-L92

Reply:
yeah, this is the chainID on the EVM side. It is a big.Int